### PR TITLE
feature/support write contract info

### PR DESCRIPTION
**Pull request by PainterPuppets**
reslove: https://github.com/ckb-js/kuai/issues/334

### CODE REVIEW

**Review by yanguoyu — APPROVED** [review]

File: packages/core/src/builtin-tasks/contract.ts [hunk 1]
```diff
@@ -213,6 +217,26 @@ subtask('contract:deploy')
...
+          hashType: hashType,
+        },
+        outPoint: {
+          txHash: txHash,
+          index: '0x' + index.toString(16),
+        },
+        depType: depType,
+      })
+
+      contractManager.write()
```
Comment (Daryl-L) [inline packages/core/src/builtin-tasks/contract.ts:239]: Use `await` maybe better I think.

**Comment by Keith-CY** [comment]
Please add a brief description or link this PR to the requirement issue in the PR message, so we can have a peek with `git log` command

File: packages/core/src/contract.ts [hunk 2]
```diff
@@ -40,6 +40,8 @@ export class ContractDeployer {
   ): Promise<{
     tx: Transaction
     index: number
+    hashType: 'type' | 'data1'
```
Comment (Keith-CY) [inline packages/core/src/contract.ts]: Why not add `data` in the type

File: packages/core/src/contract.ts [hunk 3]
```diff
@@ -104,6 +108,8 @@ export class ContractDeployer {
     tx: Transaction
     index: number
     dataHash: string
+    hashType: 'type' | 'data1'
```
Comment (Keith-CY) [inline packages/core/src/contract.ts]: Why not add `data` in the type

**Comment by Keith-CY** [comment]
Tests should be added

File: packages/core/src/builtin-tasks/contract.ts (hunk 1 above)
Comment (PainterPuppets) [reply packages/core/src/builtin-tasks/contract.ts:239]: The "write" function doesn't seem to be async function, will `await` work?

Comment (Daryl-L) [reply packages/core/src/builtin-tasks/contract.ts:239]: > The "write" function doesn't seem to be async function, will `await` work?
It's my fault, I had the wrong understanding before.

**Review by Daryl-L — APPROVED** [review]

**Pull request merged**